server/apiserver/argoserver.go

```diff
@@ -441,6 +450,67 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 	return &httpServer
 }
 
+// validateArtifactDriverImages validates that the artifact driver images are present in the server deployment
+func (as *argoServer) validateArtifactDriverImages(ctx context.Context, config *config.Config) error {
+	if len(config.ArtifactDrivers) == 0 {
+		log.Info("No artifact drivers configured, skipping validation")
+		return nil
+	}
+
+	log.Info("Validating artifact driver images against server deployment")
+
+	// Get the current pod to check what images are available
+	podName := envutil.GetHostname()
+	if podName == "" {
+		log.Warn("HOSTNAME environment variable not set, cannot validate artifact driver images")
+		return nil
+	}
+
+	// Get the current pod to check the deployment
+	pod, err := as.clients.Kubernetes.CoreV1().Pods(as.namespace).Get(ctx, podName, metav1.GetOptions{})
+	if err != nil {
+		log.Warnf("Failed to get current pod %s: %v, cannot validate artifact driver images", podName, err)
+		return nil
+	}
+
+	// Get the deployment that owns this pod
+	deployment, err := kube.GetDeploymentForPod(ctx, as.clients.Kubernetes, as.namespace, pod)
+	if err != nil {
+		log.Warnf("Failed to get deployment for pod: %v, cannot validate artifact driver images", err)
+		return nil
+	}
+	if deployment == nil {
+		log.Warn("Could not determine deployment from pod, cannot validate artifact driver images")
+		return nil
+	}
+
+	// Collect all images from the deployment
+	availableImages := make(map[string]bool)
+	for _, container := range deployment.Spec.Template.Spec.Containers {
```

Comment:
@Joibel - should I just be checking the init containers? I'm not super familiar with how sidecars would work in this context. The [docs](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/#pod-sidecar-containers) say that sidecars are defined as init containers, but based on our discussion, it seems like these plugins would be alive alongside the main container